### PR TITLE
Adds the ability to embed the course stats

### LINKED ISSUE #2233: Provide a way to embed a widget of stats from a course into another website
The main set of course stats that is shown at the top of a course page is useful info for program organizers who are showing off the impact of their programs. Users sometimes take screenshots so that they can highlight these stats other places. It would be useful if, instead, they could use a snippet of code to embed a widget on an arbitrary site that renders the latest stats for a given course.

I'm not at all sure what the best way to do that is.

### PR DESCRIPTION
**Pull request by AmitJoki**
Fixes #2233

### CODE REVIEW

**Comment by ragesoss** [comment]
I tried this out, but I get blocked by CORS whenever I try to embed it on another website. It did work when I embedded it on a different page within the development environment.

I have some questions and ideas related to the implementation...

Doing it as an iframe seems fine. Are there any other options that are worth considering, like just using a `script` tag for the embed? I think that's possible, I'm just not sure how complex it is.

I don't think sending the entire main bundle makes sense just to render a single, static component. If we're going to do it as an iframe, we probably don't even need javascript at all. The campaign stats, which look the same style-wise, get rendered by the `courses/_header_stats.html.haml` template. That approach could be used here as well, with a new haml template mostly copied from that one. Going with that approach, we'd probably also want to have the stats embed have its own route and controller that doesn't interact with the existing CoursesController and routing.

Basically, my biggest priority for how this gets implemented is to isolate it from the rest of the application as much as possible.

**Comment by ragesoss** [comment]
I get this when I try to embed it on (for example) travis-ci.org or wikiedu.org.

```
Blocked loading mixed active content “http://localhost:3000/stats/courses/University_of_Chicago/History_of_Skepticism_(Spring_2018)”
```

**Comment by ragesoss** [comment]
This looks good code-wise. The one thing I'd like to change to further isolate it is to put it in its own controller instead of in CoursesController (and put the view in its own directory instead of `views/courses`).

File: app/assets/javascripts/components/overview/available_actions.jsx [hunk 1]
```diff
@@ -189,6 +190,9 @@ const AvailableActions = createReactClass({
       controls.push((
         <p key="download_course_stats"><CourseStatsDownloadModal course={course} /></p>
       ));
+      controls.push((
+        <EmbedStatsButton title={course.title} key="embed_course_stats"/>
```
Comment (ragesoss) [inline app/assets/javascripts/components/overview/available_actions.jsx]: Needs to be wrapped in a `<p>` to the margin is right:

button margin

File: app/assets/javascripts/components/overview/embed_stats_button.jsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+          />
+          <small>{this.state.status}</small>
+        </td>
+      </tr>
+    );
+
+    return (
+      <div className="pop__container embed_stats">
+        <a onClick={this.props.open} className="button">
+          Embed Course Stats&nbsp;&lt;/&gt;
```
Comment (ragesoss) [inline app/assets/javascripts/components/overview/embed_stats_button.jsx]: i18n

File: config/locales/en.yml [hunk 3]
```diff
@@ -433,6 +433,9 @@ en:
     duration: Course Duration
     edit_count: Edit Count
     edit_course_dates: Edit Course Dates
+    embed_course_stats: Embed Course Stats
+    embed_course_stats_heading: To Embed Course Stats
```
Comment (ragesoss) [inline config/locales/en.yml]: Maybe instead of a header and then more copy, just the full explanation without a header would be clearer:

`To embed the stats for this program in another website, copy and paste this code snippet:`

**Comment by AmitJoki** [comment]
Have pushed the requested changes.

**Comment by ragesoss** [comment]
Okay! I'll merge and deploy, and then we'll see if we can get some feedback.
